### PR TITLE
feat: Preload artwork placeholder image size

### PR DESCRIPTION
**Pull request by olerichter00**
The type of this PR is: **Feature**

### Description

Currently, the dimensions we set for the artwork image placeholder don't match the artwork dimensions. This PR improves how we calculate the dimension for the placeholder by looking for the artwork image record in the Relay store. This works in most places because we usually either already fetched the width and height or the image ratio of the artwork image.

How to access the Relay store directly is influenced by [this blog post](https://yashmahalwal.medium.com/a-deep-dive-into-the-relay-store-9388affd2c2b). 

I've added [these two](https://github.com/artsy/eigen/pull/5967/files#diff-9d0f2f55f3139b0bb2b677f9902e6e0d7c27196c113ff634fdaed962f57117d6R1) helper functions to find records in the Relay store.

**Before:**
https://user-images.githubusercontent.com/4691889/147924816-6c4be1a7-498a-4721-8a0a-dae36981e6ff.mp4 

**After:**
https://user-images.githubusercontent.com/4691889/147924823-6c7e1365-3fec-4b6c-bd8b-021b9ef184cd.mp4 

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [x] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Get correct image ratio for artwork image placeholder - ole

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  ❌ **An error occurred while validating your changelog, please make sure you provided a valid changelog.**
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 5ca63824acaabd80e3b19a9363dd322c33b12e49
</p>

File: src/lib/Scenes/Artwork/Artwork.tsx [hunk 1]
```diff
@@ -526,40 +530,44 @@ export const ArtworkQueryRenderer: React.FC<{
...
 }
 
-const AboveTheFoldPlaceholder: React.FC<{}> = ({}) => {
+const AboveTheFoldPlaceholder: React.FC<{ artworkID?: string }> = ({ artworkID }) => {
   const space = useSpace()
-  const screenDimensions = useScreenDimensions()
-  // The logic for artworkHeight comes from the zeplin spec https://zpl.io/25JLX0Q
-  const artworkHeight = screenDimensions.width >= 375 ? 340 : 290
+
+  const { width, height } = useImagePlaceholderDimensions(artworkID)
```
Comment (olerichter00) [inline src/lib/Scenes/Artwork/Artwork.tsx:536]: `useImagePlaceholderDimensions` either calculates the dimensions from the artwork image (if it is already in the store) or falls back to how we calculated the dimensions in the past.

**Review by MounirDhahri — DISMISSED** [review]
I love this PR. Thanks Ole 🎉

**Comment by olerichter00** [comment]
#squashongreen

**Pull request merged**